srtcore/channel.cpp

```diff
@@ -226,31 +226,31 @@ void CChannel::setUDPSockOpt()
       }
 #endif
 
+
+#ifdef UNIX
+   // Set non-blocking I/O
+   // UNIX does not support SO_RCVTIMEO
+   int opts = ::fcntl(m_iSocket, F_GETFL);
+   if (-1 == ::fcntl(m_iSocket, F_SETFL, opts | O_NONBLOCK))
+      throw CUDTException(MJ_SETUP, MN_NORES, NET_ERROR);
+#elif defined(_WIN32)
+   u_long nonBlocking = 1;
+   if (0 != ioctlsocket (m_iSocket, FIONBIO, &nonBlocking))
+      throw CUDTException (MJ_SETUP, MN_NORES, NET_ERROR);
+#else
    timeval tv;
    tv.tv_sec = 0;
-   #if defined (BSD) || defined (OSX) || (TARGET_OS_IOS == 1) || (TARGET_OS_TV == 1)
-      // Known BSD bug as the day I wrote this code.
-      // A small time out value will cause the socket to block forever.
-      tv.tv_usec = 10000;
-   #else
-      tv.tv_usec = 100;
-   #endif
-
-   #ifdef UNIX
-      // Set non-blocking I/O
-      // UNIX does not support SO_RCVTIMEO
-      int opts = ::fcntl(m_iSocket, F_GETFL);
-      if (-1 == ::fcntl(m_iSocket, F_SETFL, opts | O_NONBLOCK))
-         throw CUDTException(MJ_SETUP, MN_NORES, NET_ERROR);
-   #elif defined(_WIN32)
-      u_long nonBlocking = 1;
-      if (0 != ioctlsocket (m_iSocket, FIONBIO, &nonBlocking))
-         throw CUDTException (MJ_SETUP, MN_NORES, NET_ERROR);
-   #else
-      // Set receiving time-out value
-      if (0 != ::setsockopt(m_iSocket, SOL_SOCKET, SO_RCVTIMEO, (char *)&tv, sizeof(timeval)))
-         throw CUDTException(MJ_SETUP, MN_NORES, NET_ERROR);
-   #endif
+#if defined (BSD) || defined (OSX) || (TARGET_OS_IOS == 1) || (TARGET_OS_TV == 1)
```

Comment:
Why do we use SO_RCVTIMEO for BSD and OSX? Both are capable of F_SETFL with O_NONBLOCK and select(). Same goes for Android. Not certain about IOS/TV.

Reply:
Also not sure. Probably it is worth to open a separate issue for that, because it requires a separate research on the topic. At least we should check and test this for OSX.

Reply:
I agree that the work on unifying approaches for all platforms can be moved out to a separate issue/PR. This PR is all about fixing breakage on Windows. Let's open a ticket and quote here.

Reply:
This would require a time consuming research from me. So I will put this to a separate issue, because we need to fix the Windows issue first. The behavior for the rest cases stays the same for now.

Reply:
This better keep focusing on just fixing the problem on Windows. We'll deal with this problem another time.